superset/connectors/sqla/models.py

```diff
@@ -1061,23 +1061,34 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                     target_column_is_numeric=col_obj.is_numeric,
                     is_list_target=is_list_target,
                 )
-                if op in (
-                    utils.FilterOperator.IN.value,
-                    utils.FilterOperator.NOT_IN.value,
-                ):
-                    cond = col_obj.get_sqla_col().in_(eq)
-                    if isinstance(eq, str) and NULL_STRING in eq:
```

Comment:
`NULL_STRING` is already parsed to `None` in `filter_values_handler`, so this is not needed.